core/src/main/resources/digester-rules.xml

```diff
@@ -3,27 +3,26 @@
 	<!-- Rules are parsed with the DigesterRuleParser. Objects will be created with the Spring Beans Factory. -->
 	<!-- See DigesterRule for more information -->
 
-	<rule pattern="IOS-Adaptering" factory="null"/><!-- deprecated, use configuration or module instead -->
 	<rule pattern="configuration" factory="null"/>
 
 	<rule pattern="*/include" factory="nl.nn.adapterframework.configuration.digester.ConfigurationDigesterFactory" selfRegisterMethod="include"/>
 
+	<rule pattern="configuration/jmsRealms" factory="nl.nn.adapterframework.configuration.digester.JmsRealmsFactory"/>
+	<rule pattern="*/jmsRealm" registerMethod="registerJmsRealm" beanClass="nl.nn.adapterframework.jms.JmsRealm"/>
 
```

Comment:
Dit voorstel introduceert complexiteit in de Frank!Doc, zoals telefonisch besproken met @nielsm5. De "jmsRealms" rule noemt geen Java class. De Frank!Doc bouwt nu config children door een ElementType te creëren uit een Java class of interface. Als het ElementType een interface is, dan zijn de toegestane kinderen alle implementaties van het interface. Anders is de class van het ElementType het enige toegestane kind.

Ik kan de volgende truuk bedenken. We introduceren een nieuw soort config child dat geen ElementType heeft. Dit type config child presenteert zichzelf als "container voor". Dus in de XSDs en in de JSON komt als documentatie te staan dat `<jmsRealms>` de container is voor `<jmsRealm>`. Hetzelfde kunnen we doen voor `<sapSystems>` en `<events>` betreffende monitoring. Ik denk dat het te doen is om dit te implementeren. Het is ongeveer even moeilijk als het was om TextConfigChild te introduceren voor monitoring.

Het nadeel van deze aanpak is dat onze taal niet consistent in het wel/niet voorschrijven van containers. Forwards en exits bestaan los, dus niet `<forwards><forward>...`. Maar wel `<jmsRealms><jmsRealm>...`.

Reply:
We zitten ook nog met volgorde. Als we config children maken die niet corresponderen met een register method in een Java class, hoe bepalen we dan de volgorde? Moeten alle 'losse' config children vooraaan of achteraan? En kunnen de losse config children gesorteerd worden corresponderend met de volgorde in `digester-rules.xml`? Of hebben we een extra attribute nodig in `digester-rules.xml` voor de volgorde?

Reply:
We kunnen ook een Java class introduceren waar de documentatie voor `<jmsRealms>` in staat, bijvoorbeeld `FrankDocJmsRealmsContainer`. Het F!F gebruikt deze class niet, maar hij biedt wel plaats aan een method `addJmsRealms()`. De JavaDoc documenteert config child `<jmsRealms>` van `Configuration` in de JSON. Misschien is het ook handig om een Java class `JmsRealms` te hebben die method `registerJmsRealm()` heeft. De JavaDoc van methld `JmsRealms.registerJmsRealm()` bevat dan de documentatie van config child `<jmsRealm>` in de JSON.

Reply:
Als we deze classes introduceren en ernaar verwijzen in `digester-rules.xml`, dan hoef ik minder te doen aan de Frank!Doc. Er is dan geen nieuw type ConfigChild nodig. Het is nog wel complex dat er twee parent Java classes zijn van config child `<jmsRealms>`. Zowel Configuration als FrankDocJmsRealmsContainer spelen die rol.